File: 백준/Bronze/2991. 사나운 개/사나운 개.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');

const [A, B, C, D] = input[0].split(' ').map(Number);
const deliveryTimes = input[1].split(' ').map(Number);

function isAttacked(time, attack, rest) {
    const period = attack + rest;
    const status = (time - 1) % period;
    return status < attack ? 1 : 0;
}

deliveryTimes.forEach(time => {
    let count = 0;
    count += isAttacked(time, A, B);
    count += isAttacked(time, C, D);
    console.log(count);
});

File: 백준/Bronze/2991. 사나운 개/사나운 개.py
A, B, C, D = map(int, input().split())
arrival_times = list(map(int, input().split()))

for time in arrival_times:
    count = 0

    if (time - 1) % (A + B) < A:
        count += 1

    if (time - 1) % (C + D) < C:
        count += 1
    
    print(count)